python/ray/data/dataset.py

```diff
@@ -785,6 +796,7 @@ def flat_map(
         num_cpus: Optional[float] = None,
         num_gpus: Optional[float] = None,
         concurrency: Optional[Union[int, Tuple[int, int]]] = None,
+        scheduling_strategy_fn: Optional[Callable[[], "SchedulingStrategyT"]] = None,
```

Comment:
discussed with @c21 offline, it would be more generalizable to add a `ray_remote_args_fn`, because we may want to dynamically generate other args as well. 
The semantics can be that if both the ray_remote_args_fn and other kwargs ray_remote_args are specified, the kwargs take higher priority.

Reply:
also, this usage is a bit too advanced, maybe let's mark it as experimental for now